@here/harp-mapview/lib/Utils.ts

```diff
@@ -446,6 +460,229 @@ export namespace MapViewUtils {
+                    }
+                }
+            }
+        } else if (
+            material instanceof THREE.MeshBasicMaterial ||
+            material instanceof MapMeshBasicMaterial
+        ) {
+            const meshMaterial = material;
+            estimateTextureSize(meshMaterial.map, isVisible, objectSize);
+            estimateTextureSize(meshMaterial.aoMap, isVisible, objectSize);
+            estimateTextureSize(meshMaterial.specularMap, isVisible, objectSize);
+            estimateTextureSize(meshMaterial.alphaMap, isVisible, objectSize);
+            estimateTextureSize(meshMaterial.envMap, isVisible, objectSize);
+        } else if (material instanceof MapMeshStandardMaterial) {
+            const standardMaterial = material;
+            estimateTextureSize(standardMaterial.map, isVisible, objectSize);
+            estimateTextureSize(standardMaterial.aoMap, isVisible, objectSize);
+            estimateTextureSize(standardMaterial.alphaMap, isVisible, objectSize);
+            estimateTextureSize(standardMaterial.envMap, isVisible, objectSize);
+        } else if (
+            material instanceof THREE.LineBasicMaterial ||
+            material instanceof THREE.LineDashedMaterial
+        ) {
+            // Nothing to be done here
+        } else {
+            logger.warn("estimateMeshSize: unidentified material: ", material);
+        }
+    }
+
+    function estimateAttributeSize(
+        attribute: any,
+        attrName: string,
+        isVisible: boolean,
+        objectSize: MemoryUsage
+    ) {
+        let bytesPerElement = 4;
+        let attrBytes = 0;
+        if (attribute.array.BYTES_PER_ELEMENT !== undefined) {
+            bytesPerElement = attribute.array.BYTES_PER_ELEMENT;
+        }
+        if (attribute instanceof THREE.InterleavedBufferAttribute) {
+            const interAttr = attribute;
+            if (interAttr.offset !== 0) {
+                // Array is shared between attributes, count it only once
+                return;
+            }
+            attrBytes = bytesPerElement * attribute.count * attribute.itemSize;
+        } else if (attribute instanceof THREE.BufferAttribute) {
+            const bufferAttr = attribute;
+            attrBytes = bytesPerElement * bufferAttr.count * bufferAttr.itemSize;
+        } else {
+            logger.warn("estimateMeshSize: unidentified attribute: ", attrName);
+        }
+
+        objectSize.heapSize += attrBytes;
+        if (isVisible) {
+            objectSize.gpuSize += attrBytes;
+        }
+    }
+
+    /**
+     * Computes estimate for size of a single THREE.Mesh or THREE.Line object. Shared materials
+     * and/or attributes will be counted multiple times.
+     *
+     * @param object The mesh object to evaluate
+     * @param size The [[MemoryUsage]] to update.
+     */
+    export function estimateMeshSize(object: THREE.Object3D, objectSize: MemoryUsage): void {
```

Comment:
Without `ArrayBuffer` tracing, this one in particular with double memory usage for any re-used geometry like extruded buildings depth and pre-pass or future line foreground and background.

Again, this may be minor and room for enchancement.

Reply:
I will add code to count attributes/arraybuffers and textures only once. (see above) Counting the buffers twice for extruded buildings with outlines is actually really significant.

Reply:
Added handling of shared Geometries, Materials, Textures and Attributes.